Calculator.Logic/CalculatorLogic.cs

```diff
@@ -0,0 +1,323 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Calculator.Logic
+{
+    public class CalculatorLogic : ICalculatorLogic
+    {
+        private ICalculatorError _calculatorError;
+        private ICalculatorValidator _calculatorValidator;
+        private IStringToNumberConvertor _stringToNumberConverter;
+        private List<string> _terms;
+        private Operation _currentOperation;
+        private string _lastTerm;
+
+        public CalculatorLogic(ICalculatorError calculatorError, ICalculatorValidator calculatorValidator, IStringToNumberConvertor stringToNumberConverter)
+        {
+            _calculatorError = calculatorError;
+            _calculatorValidator = calculatorValidator;
+            _stringToNumberConverter = stringToNumberConverter;
+            _terms = new List<string>();
+        }
+       
+        public CalculatorResult DoMultipleTermOperation()
+        {
+            if (ShouldRepeatLastOperation(_currentOperation))
+            {
+                AddLastTermUsed(_lastTerm);
+            }
+            try
+            {
+                return DoCalculation();
+            }
+            catch (CalculatorException ex)
+            {
+                return _calculatorError.HandleCalculatorException(ex);
+            }
+        }
+
+        public CalculatorResult DoCalculation()
+        {
+            _lastTerm = _terms.LastOrDefault();
+            _calculatorValidator.ValidateOperation(_currentOperation);
+            CalculatorResult calculatorResult = null;
+            switch (_currentOperation)
+            {
+                case Operation.Sum:
+                    {
+                        List<double> termsList = GetTerms();
```

Comment:
Extract method for Operations Sum,Substract,Multiply,Divide,Power that thakes a Func<double, double, double> as operation and a Action<List<double>> that takes a ValidationMethod

Reply:
![image](https://user-images.githubusercontent.com/96069075/221411578-c990e1f1-a8a8-4f25-af47-533b3e05e758.png)

Reply:
![image](https://user-images.githubusercontent.com/96069075/221436928-82abba29-7fea-42eb-b8a3-00170889913a.png)

Reply:
![image](https://user-images.githubusercontent.com/96069075/221436970-adfbf14c-e441-40a9-aedf-3c665b9885a0.png)